lib/spack/llnl/util/tty/__init__.py

```diff
@@ -26,7 +26,7 @@
 _warn_enabled = True
 _error_enabled = True
 indent = "  "
-_osascript = os.path.exists('/usr/bin/osascript')
+_osascript = subprocess.call(['which', 'osascript']) == 0
```

Comment:
Much cleaner. 👍

Reply:
@adamjstewart @tldahlgren: only thing about this code is that it depends on `which`, and I kind of want to avoid having external command dependencies in this code.

The rest of spack uses `spack.util.executable.which` for things like this, but that returns a `spack.util.executable.Executable`, which is (kind of) spack-specific.

In #11245 I pulled out `which_string` from `which()`, that searches the path and gives you a string if the thing exists and is executable.  How about we pull that in as `llnl.util.filesystem.which`, and use it instead of shelling out to `which`?

Alternately, `spack.util.executable` is not THAT spack-specific.  The only thing that you'd have to change to move it to `llnl.util` would be to rethink `ProcessError`, which extends `SpackError`, which a lot of parts of Spack use for error handling.  Probably not worth dealing with that now -- but it's something we could do later.  Just using `which_string` is probably fine for now.

Reply:
We can't use `which_string` because `spack.util.executable` imports `llnl.util.tty`. We would either have to reimplement it in `tty` or remove the `tty` imports.

Reply:
I meant *move* the function to llnl.util.

Reply:
Are we still thinking about merging `llnl` and `spack`? Also, did you have any opinions on merging `tty.msg` and `tty.info`?

Reply:
Not at the moment, though I have thought about making some things in llnl.util into proper Python packages.

Reply:
I think you can merge `info` into `msg`.  I have no recollection at this point of why they're different.

Reply:
Did you want to keep `info` or `msg`? I'm leaning towards `info` since it's more inline with https://docs.python.org/3/library/logging.html#logging-levels. Also, should we have a command-line or config option to enable/disable marks?

Reply:
Also, `tty.verbose` doesn't seem to be used anywhere? Is this correct?